File: backend/components/metiers/threed/api/samples/validators/sample_validators.js
/* global console */
// Exemple ultra avancé clé en main des validateurs API Threed (JS)
const { validate3dEntity } = require('../validators/validators');
const { auditEntity } = require('../audit/audit');

function sampleValidatorsUltra() {
  // Validation standard
  validate3dEntity({ name: 'Test', status: 'active' });
  auditEntity({ name: 'Test' }, 'validate');
  console.info('Validation standard OK');
  // Edge case: entité incomplète
  try {
    validate3dEntity({ status: 'active' });
  } catch (e) {
    console.warn('Validation échouée:', e);
  }
  console.log('Validateurs ultra avancé exécuté avec succès.');
}

module.exports = { sampleValidatorsUltra };

File: backend/components/metiers/threed/api/samples/validators/sample_validators.py
# Exemple ultra avancé clé en main des validateurs API Threed (Python)
from ..validators.validators import validate_3d_entity
from ..audit.audit import audit_entity
import logging

def sample_validators_ultra():
    # Validation standard
    validate_3d_entity({'name': 'Test', 'status': 'active'})
    audit_entity({'name': 'Test'}, 'validate')
    logging.info('Validation standard OK')
    # Edge case: entité incomplète
    try:
        validate_3d_entity({'status': 'active'})
    except Exception as e:
        logging.warning(f"Validation échouée: {e}")
    print('Validateurs ultra avancé exécuté avec succès.')
